client/vmmCtrlrP_service.go

```diff
@@ -0,0 +1,259 @@
+package client
+
+import (
+	"fmt"
+
+	"github.com/ciscoecosystem/aci-go-client/container"
+	"github.com/ciscoecosystem/aci-go-client/models"
+)
+
+func (sm *ServiceManager) CreateVMMController(name string, vmm_domain string, provider_profile_vendor string, description string, vmmCtrlrPattr models.VMMControllerAttributes) (*models.VMMController, error) {
+	rn := fmt.Sprintf("ctrlr-%s", name)
+	parentDn := fmt.Sprintf("uni/vmmp-%s/dom-%s", provider_profile_vendor, vmm_domain)
+	vmmCtrlrP := models.NewVMMController(rn, parentDn, description, vmmCtrlrPattr)
+	err := sm.Save(vmmCtrlrP)
+	return vmmCtrlrP, err
+}
+
+func (sm *ServiceManager) ReadVMMController(name string, vmm_domain string, provider_profile_vendor string) (*models.VMMController, error) {
+	dn := fmt.Sprintf("uni/vmmp-%s/dom-%s/ctrlr-%s", provider_profile_vendor, vmm_domain, name)
+	cont, err := sm.Get(dn)
+	if err != nil {
+		return nil, err
+	}
+
+	vmmCtrlrP := models.VMMControllerFromContainer(cont)
+	return vmmCtrlrP, nil
+}
+
+func (sm *ServiceManager) DeleteVMMController(name string, vmm_domain string, provider_profile_vendor string) error {
+	dn := fmt.Sprintf("uni/vmmp-%s/dom-%s/ctrlr-%s", provider_profile_vendor, vmm_domain, name)
+	return sm.DeleteByDn(dn, models.VmmctrlrpClassName)
+}
+
+func (sm *ServiceManager) UpdateVMMController(name string, vmm_domain string, provider_profile_vendor string, description string, vmmCtrlrPattr models.VMMControllerAttributes) (*models.VMMController, error) {
+	rn := fmt.Sprintf("ctrlr-%s", name)
+	parentDn := fmt.Sprintf("uni/vmmp-%s/dom-%s", provider_profile_vendor, vmm_domain)
+	vmmCtrlrP := models.NewVMMController(rn, parentDn, description, vmmCtrlrPattr)
+
+	vmmCtrlrP.Status = "modified"
+	err := sm.Save(vmmCtrlrP)
+	return vmmCtrlrP, err
+
+}
+
+func (sm *ServiceManager) ListVMMController(vmm_domain string, provider_profile_vendor string) ([]*models.VMMController, error) {
+
+	baseurlStr := "/api/node/class"
+	dnUrl := fmt.Sprintf("%s/uni/vmmp-%s/dom-%s/vmmCtrlrP.json", baseurlStr, provider_profile_vendor, vmm_domain)
+
+	cont, err := sm.GetViaURL(dnUrl)
+	list := models.VMMControllerListFromContainer(cont)
+
+	return list, err
+}
+
+func (sm *ServiceManager) CreateRelationvmmRsMcastAddrNs(parentDn, tnFvnsMcastAddrInstPName string) error {
+	dn := fmt.Sprintf("%s/rsmcastAddrNs", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s",
+				"tnFvnsMcastAddrInstPName": "%s"
```

Comment:
We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.